parts/k8s/windowsazurecnifunc.ps1

```diff
@@ -332,15 +332,36 @@ function New-ExternalHnsNetwork
 
     Write-Log "Creating new HNS network `"ext`""
     $externalNetwork = "ext"
-    $na = @(Get-NetAdapter -Physical)
+    $nas = @(Get-NetAdapter -Physical)
 
-    if ($na.Count -eq 0) {
+    if ($nas.Count -eq 0) {
         throw "Failed to find any physical network adapters"
     }
 
-    # If there is more than one adapter, use the first adapter.
-    $managementIP = (Get-NetIPAddress -ifIndex $na[0].ifIndex -AddressFamily IPv4).IPAddress
-    $adapterName = $na[0].Name
+    # If there is more than one adapter, use the first adapter that is assigned an ipaddress.
+    foreach($na in $nas)
+    {
+        $netIP = Get-NetIPAddress -ifIndex $na.ifIndex -AddressFamily IPv4 -ErrorAction SilentlyContinue -ErrorVariable a

```

Comment:
nit, `err` or `netIPErr` may be more descriptive names

Reply:
fixed